controllers/storagecluster/cephcluster.go

```diff
@@ -882,7 +883,7 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rookCephv1.StorageCla
 				Portable:            portable,
 				TuneSlowDeviceClass: ds.Config.TuneSlowDeviceClass,
 				TuneFastDeviceClass: ds.Config.TuneFastDeviceClass,
-				Encrypted:           sc.Spec.Encryption.Enable || sc.Spec.Encryption.ClusterWide,
+				Encrypted:           sc.Spec.Encryption.Enable || sc.Spec.Encryption.ClusterWide || ds.Encrypted,
```

Comment:
we need a fallback mechanism here,
We don't have just trust `true`

If the deviceset is set to `true` or `false` we should use it.
If not then we should see the storagecluster top-level `Encryption` values

Reply:
But if someone has specified cluster wide encryption shouldn't that have more precedence over what the individual devicseSet is specifying?

Reply:
Added the new logic as discussed in the call